qutebrowser/misc/editor.py

```diff
@@ -121,21 +122,93 @@ def edit(self, text):
         self._proc.finished.connect(self.on_proc_closed)
         self._proc.error.connect(self.on_proc_error)
         editor = config.val.editor.command
         executable = editor[0]
-        args = [arg.replace('{}', self._filename) for arg in editor[1:]]
+
+        args = [self._sub_placeholder(arg, line, column) for arg in editor[1:]]
         log.procs.debug("Calling \"{}\" with args {}".format(executable, args))
         self._proc.start(executable, args)
+
+    def _calc_line_and_column(self, text, caret_position):
+        """Calculate line and column numbers given a text and caret position
+
+        Args:
+            text: the text for which the numbers must be calculated
+            caret_position: the position of the caret in the text
+
+        Return:
+            A (line, column) tuple of (int, int)
+        """
+        # Both line and column are 1-based indexes, because that's what most
+        # editors use as line and column starting index.  By "most" we mean at
+        # least vim, nvim, gvim, emacs, atom, sublimetext, notepad++, brackets,
+        # visual studio, QtCreator and so on.
+        #
+        # To find the line we just count how many newlines there are before the
+        # caret and add 1.
+        #
+        # To find the column we calculate the difference between the caret and
+        # the last newline before the caret.
+        #
+        # For example in the text `aaa\nbb|bbb` (| represents the caret):
+        # caret_position = 6
+        # text[:caret_position] = `aaa\nbb`
+        # text[:caret_psotion].count('\n') = 1
+        # caret_position - text[:caret_position].rfind('\n') = 3
+        #
+        # Thus line, column = 2, 3, and the caret is indeed in the second
+        # line, third column
+        line = text[:caret_position].count('\n') + 1
+        column = caret_position - text[:caret_position].rfind('\n')
+        return (line, column)
+
+    def _sub_placeholder(self, possible_placeholder, line, column):
+        """Substitute a single placeholder.
+
+        If the `possible_placeholder` input to this function is valid it will
+        be substituted with the appropriate value, otherwise it will be left
+        unchanged.
+
+        Args:
+            possible_placeholder: an argument of editor.command.
+            line: the previously-calculated line number for the text caret
+            column: the previously-calculated column number for the text caret
+
+        Return:
+            The substituted placeholder or the original argument
+        """
+        replacements = {
+            '{}': self._filename,
+            '{file}': self._filename,
+            '{line}': str(line),
+            '{line0}': str(line-1),
+            '{column}': str(column),
+            '{column0}': str(column-1)
+        }
+
+        # The for loop would create a copy anyway, this should be more legible
+        ph = possible_placeholder
```

Comment:
Maybe just rename the `possible_placeholder` argument to `arg` (as in editor argument) or so? After all, it can also be `foo{placeholder}bar` or so.

(Also, note that neither is a copy - just [different references to the same object](http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html#other-languages-have-variables) :wink:)